dask/base.py

```diff
@@ -987,6 +990,37 @@ def _normalize_function(func):
             return str(func)
 
 
+def normalize_dataclass(obj):
+    # If dataclasses should not be compared, they should not be deterministically
+    # tokenized either, since the purpose of tokenization is comparision.
+    try:
+        comparable = obj.__dataclass_params__.eq
+    except AttributeError:
+        comparable = False
+
+    if comparable:
+        fields = [
+            (field.name, getattr(obj, field.name))
+            for field in dataclasses.fields(obj)
+            if field.compare
+        ]
+        if fields:
+            return (
+                normalize_function(type(obj)),
+                _normalize_seq_func(fields),
```

Comment:
Typically we use `normalize_token`, and let it handle the appropriate type dispatching, instead of calling specific methods like `_normalize_seq_func`. This is a minor point, but is nice because it's resilient to future tweaks in the names of internal functions like `_normalize_seq_func` 

```suggestion
                normalize_token(fields),
```

Reply:
This was purely a performance optimization in order to not have to dispatch on the list type. It felt reasonable to me as an "internal function calling other internal function" case, but I can change it. (Same reason I used `normalize_function` directly above, as well).

Reply:
Calling the direct function seems fine, up to you.